spec/integration/util/windows/principal_spec.rb

```diff
@@ -111,5 +142,12 @@
         principal.lookup_account_sid([1, 1, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0])
       }.to raise_error(Puppet::Util::Windows::Error, /Failed to call LookupAccountSidW:  No mapping between account names and security IDs was done/)
     end
+    
+    it "should return a domain principal for BUILTIN SID S-1-5-32" do
+      principal = Puppet::Util::Windows::SID::Principal.lookup_account_sid([1, 1, 0, 0, 0, 0, 0, 5, 32, 0, 0, 0])
+      expect(principal.account_type).to eq(:SidTypeDomain)
+      expect(principal.sid).to eq('S-1-5-32')
+      expect(principal.domain).to eq('BUILTIN')
+    end    
```

Comment:
Ah yeah, but what about looking up `BUILTIN` by name?  we should see that `account_domain` is hosed as `BUILTIN\BUILTIN` - which is the part I was interested in fixing now (one extra commit is fine to address that)